pymc_experimental/model/transforms/autoreparam.py

```diff
@@ -175,14 +177,13 @@ def vip_reparam_node(
     if not isinstance(node.op, RandomVariable | SymbolicRandomVariable):
         raise TypeError("Op should be RandomVariable type")
     rv = node.default_output()
-    try:
-        [rv_shape] = constant_fold([rv.shape])
-    except NotConstantValueError:
-        raise ValueError("Size should be static for autoreparametrization.")
+    [rv_shape] = constant_fold([rv.shape], raise_if_not_constant=False)
```

Comment:
Ignore my idea, this won't work for you. When it can't constant_fold, it returns the best graph it can but you still need constants for `np.zeros`